src/native/corehost/hostmisc/pal.h

```diff
@@ -287,7 +287,7 @@ namespace pal
     // Returns the globally registered install location (if any)
     bool get_dotnet_self_registered_dir(string_t* recv);
     // Returns name of the global registry location (for error messages)
-    bool get_dotnet_self_registered_config_location(string_t* recv);
+    void get_dotnet_self_registered_config_location(string_t* recv);
```

Comment:
Would it make sense to have this just return `pal::string_t` now instead of taking the pointer?

Reply:
Yes. Taking a pointer makes things complicated. Returning a `pal::string_t` makes much more sense and is more efficient.